packages/core/admin/admin/src/content-manager/pages/ListConfiguration/ListConfigurationPage.tsx

```diff
@@ -0,0 +1,208 @@
+import { SortDisplayedFields } from './components/SortDisplayedFields';
+
+import type { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
+
+const EXCLUDED_SORT_ATTRIBUTE_TYPES = [
+  'media',
+  'richtext',
+  'dynamiczone',
+  'relation',
+  'component',
+  'json',
+  'blocks',
+];
+
+interface FormData extends Pick<ListLayout, 'settings'> {
+  layout: Array<Pick<ListFieldLayout, 'sortable' | 'name'> & { label: string }>;
+}
+
+const ListConfiguration = () => {
+  const { formatMessage } = useIntl();
+  const { trackUsage } = useTracking();
+  const toggleNotification = useNotification();
+  const { _unstableFormatAPIError: formatAPIError } = useAPIErrorHandler();
+
+  const { model, collectionType, schema } = useDoc();
+
+  const { isLoading: isLoadingLayout, list, edit } = useDocLayout();
+
+  const [updateContentTypeConfiguration] = useUpdateContentTypeConfigurationMutation();
+  const handleSubmit: FormProps<FormData>['onSubmit'] = async (data, event) => {
+    event.preventDefault();
+
+    try {
+      trackUsage('willSaveContentTypeLayout');
+
+      /**
+       * We reconstruct the metadatas object by taking the existing edit metadatas
+       * and re-merging that by attribute name with the current list metadatas, whilst overwriting
+       * the data from the form we've built.
+       */
+      const meta = Object.entries(edit.metadatas).reduce<Contracts.ContentTypes.Metadatas>(
+        (acc, [name, editMeta]) => {
+          const { mainField: _mainField, ...listMeta } = list.metadatas[name];
+
+          const { label, sortable } = data.layout.find((field) => field.name === name) ?? {};
+
+          acc[name] = {
+            edit: editMeta,
+            list: {
+              ...listMeta,
+              label,
+              sortable,
+            },
+          };
+
+          return acc;
+        },
+        {}
+      );
+
+      const res = await updateContentTypeConfiguration({
+        layouts: {
+          edit: edit.layout.flatMap((panel) =>
+            panel.map((row) => row.map(({ name, size }) => ({ name, size })))
+          ),
+          list: data.layout.map((field) => field.name),
+        },
+        settings: setIn(data.settings, 'displayName', undefined),
+        metadatas: meta,
+        uid: model,
+      });
+
+      if ('data' in res) {
+        trackUsage('didEditListSettings');
+        toggleNotification({
+          type: 'success',
+          message: { id: 'notification.success.saved', defaultMessage: 'Saved' },
+        });
+      } else {
+        toggleNotification({
+          type: 'warning',
+          message: formatAPIError(res.error),
+        });
+      }
+    } catch {
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error', defaultMessage: 'An error occurred' },
+      });
+    }
+  };
+
+  const sortOptionsCE = React.useMemo(
+    () =>
+      Object.values(list.layout)
```

Comment:
Is possible to keep `list.layout` updated with the changes from the user before save? Or use different place to get the sortOptions?

It's a really small detail, but right now if you add a new attribute to the view, to select it on the sortBy dropdown you need to save first and then it would show on the dropdown. 


https://github.com/strapi/strapi/assets/23050694/9820b97b-1f53-405f-a528-7143307f8c21